### PR TITLE
Create UserResponseDto and controller with methods

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

File: src/main/java/booking/config/AppConfig.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 @Configuration
 @PropertySource("classpath:db.properties")
-@ComponentScan(basePackages = {"booking.service", "booking.dao"})
+@ComponentScan(basePackages = {"booking.service", "booking.dao", "b"})
```
Comment (Max-Pochepets) [inline src/main/java/booking/config/AppConfig.java]: What is "b"?

File: src/main/java/booking/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @GetMapping("/{user_id}")
+    public UserResponseDto getUser(@PathVariable("user_id") Long id) {
+        return userMapper.getDto(userService.get(id));
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        return userService.listUsers().stream()
+                .map(userMapper::getDto)
```
Comment (Max-Pochepets) [inline src/main/java/booking/controller/UserController.java]:
```suggestion
                .map(userMapper::toDto)
```
In future you'll have RequestDto's, how would you call responsive method? `getModel`?

File: src/main/java/booking/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import booking.service.UserService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/user")
```
Comment (Max-Pochepets) [inline src/main/java/booking/controller/UserController.java]:
```suggestion
@RequestMapping("/users")
```

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/booking/config/AppConfig.java (hunk 1 above)
Comment (vicire) [reply src/main/java/booking/config/AppConfig.java]: Fixed

File: src/main/java/booking/controller/UserController.java (hunk 2 above)
Comment (vicire) [reply src/main/java/booking/controller/UserController.java]: Fixed

File: src/main/java/booking/controller/UserController.java (hunk 3 above)
Comment (vicire) [reply src/main/java/booking/controller/UserController.java]: Fixed

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/booking/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @GetMapping("/inject")
+    public String injectUser() {
+        User[] users = new User[4];
+        for (int i = 0; i < users.length; i++) {
+            users[i] = new User();
+            users[i].setName("Childish");
+            users[i].setSurname("Gambino" + i);
+            users[i].setEmail("dto@gmail.com");
```
Comment (dmytro-hryhoruk) [inline src/main/java/booking/controller/UserController.java]: email should be unique

File: src/main/java/booking/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    private final UserMapper userMapper;
+
+    public UserController(UserService userService, UserMapper userMapper) {
+        this.userService = userService;
+        this.userMapper = userMapper;
+    }
+
+    @GetMapping("/inject")
+    public String injectUser() {
+        User[] users = new User[4];
```
Comment (dmytro-hryhoruk) [inline src/main/java/booking/controller/UserController.java]: do you need array?

File: src/main/java/booking/service/UserServiceImpl.java [hunk 6]
```diff
@@ -22,4 +22,12 @@ public void add(User user) {
     public List<User> listUsers() {
         return userDao.listUsers();
     }
+
+    @Override
+    public User get(Long id) {
+        if (userDao.get(id).isEmpty()) {
+            return null;
+        }
+        return userDao.get(id).get();
```
Comment (dmytro-hryhoruk) [inline src/main/java/booking/service/UserServiceImpl.java]: You are calling db twice.
let's throw an exception if the user is null and let's do this inline

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/booking/controller/UserController.java (hunk 4 above)
Comment (vicire) [reply src/main/java/booking/controller/UserController.java]: Fixed

File: src/main/java/booking/controller/UserController.java (hunk 5 above)
Comment (vicire) [reply src/main/java/booking/controller/UserController.java]: Fixed

File: src/main/java/booking/service/UserServiceImpl.java (hunk 6 above)
Comment (vicire) [reply src/main/java/booking/service/UserServiceImpl.java]: Fixed

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**